src/chunk_insert_state.c

```diff
@@ -535,6 +549,12 @@ ts_chunk_insert_state_create(Chunk *chunk, ChunkDispatch *dispatch)
 	return state;
 }
 
+void
+ts_chunk_insert_state_switch(ChunkInsertState *state, ChunkDispatch *dispatch)
```

Comment:
see earlier comment on chunk_dispatch_state.  Could you move this code to get_chunk_insert_state and remove this one?

Reply:
Do you mean the `adjust_slots` call should be in `ts_chunk_dispatch_get_chunk_insert_state`? I think it's better to just tell the CIS that we are switching to it and have it decide what it needs to do. I think chunk_dispatch should know nothing about adjusting slots.